build/ci.yml

```diff
@@ -48,26 +49,41 @@ jobs:
   - template: steps-selfcontained.yml
   condition: ne(variables['Enable.Conda'], 'false')
 
-- job: "pack_conda_linux"
+- job: "pack_conda_macos"
   dependsOn: "pack_selfcontained"
   pool:
-    vmImage: 'ubuntu-latest'
+    vmImage: 'macOS-latest'
+  variables:
+    OS: 'mac'
+    CondaShellHook: "$('/usr/local/bin/conda' 'shell.bash' 'hook')"
   steps:
   - template: steps-conda.yml
   condition: ne(variables['Enable.Conda'], 'false')
 
-- job: "pack_conda_macos"
+- job: "pack_conda_linux"
   dependsOn: "pack_selfcontained"
   pool:
-    vmImage: 'macOS-latest'
+    vmImage: 'ubuntu-latest'
+  container:
+    image: qdkimages.azurecr.io/internal/quantum/linux-test:latest
+    endpoint: qdkimages-acr
+  variables:
+    OS: 'linux'
+    CondaShellHook: "$('/miniconda/bin/conda' 'shell.bash' 'hook')"
   steps:
   - template: steps-conda.yml
   condition: ne(variables['Enable.Conda'], 'false')
 
-- job: "pack_conda_win"
+- job: "pack_conda_windows"
   dependsOn: "pack_selfcontained"
   pool:
     vmImage: 'windows-2019'
+  container:
+    image: qdkimages.azurecr.io/internal/quantum/windows-test:latest
```

Comment:
We may want a slightly more descriptive name than `windows-test` here. It'd also be good to pick this up from the internal MCR (`msint.azurecr.io`) and not from our ACR instance that feeds into it (`qdkimages.azurecr.io`). To do so, we'll likely need a matching PR against the microsoft/mcr repo.

Reply:
I've updated the images to `linux/windows-selfcontained`. I also had the MCR PR approved below:
https://github.com/microsoft/mcr/pull/1871

Lastly I've updated the `push-selfcontained.yml` workflow to run for both linux and windows images.